connectors/byoc.py

```diff
@@ -210,6 +217,12 @@ async def update_metadata(self, ingestion_stats=None, connector_metadata=None):
         if connector_metadata is None:
             connector_metadata = {}
 
+        ingestion_stats = {
+            k: v
+            for (k, v) in ingestion_stats.items()
+            if k in ALLOWED_INGESTION_STATS_KEYS
+        }
+
```

Comment:
Might be my lack of knowledge on python standards, but is it possible to have this and the algorithm below to be in the same method so the code is in one place?